src/corehost/cli/libhost.h

```diff
@@ -194,27 +206,35 @@ struct hostpolicy_init_t
             trace::error(_X("The version of the data layout used to initialize %s is [0x%04x]; expected version [0x%04x]"), LIBHOSTPOLICY_NAME, input->version_hi, HOST_INTERFACE_LAYOUT_VERSION_HI);
             return false;
         }
-        // Check if the size is at least what we expect to contain.
-        if (input->version_lo < HOST_INTERFACE_LAYOUT_VERSION_LO)
-        {
-            trace::error(_X("The size of the data layout used to initialize %s is %d; expected at least %d"), LIBHOSTPOLICY_NAME, input->version_lo, HOST_INTERFACE_LAYOUT_VERSION_LO);
-            return false;
-        }
+
         trace::verbose(_X("Reading from host interface version: [0x%04x:%d] to initialize policy version: [0x%04x:%d]"), input->version_hi, input->version_lo, HOST_INTERFACE_LAYOUT_VERSION_HI, HOST_INTERFACE_LAYOUT_VERSION_LO);
 
-        make_clrstr_arr(input->config_keys.len, input->config_keys.arr, &init->cfg_keys);
-        make_clrstr_arr(input->config_values.len, input->config_values.arr, &init->cfg_values);
+        if (input->version_lo >= offsetof(host_interface_t, host_mode) + sizeof(input->host_mode))
```

Comment:
Pls add a comment here as per @gkhanna79 feedback in PR #1109. That the if checks are for compat between new/old fxr/policy, and we should not garbage due to potentially shorter struct size.